File: 01.scala-learning-atguigu/src/main/scala/com/yankee/chapter01/Student.scala
package com.yankee.chapter01

/**
 * @Description 学生类对象
 * @Date 2022/3/21 15:42
 * @Author yankee
 */
class Student(val name: String, var age: Int) {
  def printInfo(): Unit = {
    println(name + "  " + age + "  " + Student.school)
  }
}

/**
 * 引入伴生对象
 */
object Student {
  val school: String = "atguigu"

  def main(args: Array[String]): Unit = {
    val alice = new Student("alice", 20)
    val bob = new Student("bob", 30)
    alice.printInfo()
    bob.printInfo()
  }
}

File: 01.scala-learning-atguigu/src/main/java/com/yankee/Student.java
package com.yankee;

import scala.Int;

/**
 * @Description 学生类对象
 * @Date 2022/3/22 13:26
 * @Author yankee
 */
public class Student {
    /**
     * 姓名
     */
    private String name;

    /**
     * 年龄
     */
    private Integer age;

    /**
     * 类的静态属性
     */
    private static String school = "清华";

    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    public void printInfo() {
        System.out.println(this.name + "  " + this.age + "  " + Student.school);
    }

    public static void main(String[] args) {
        Student student = new Student("alice", 20);
        student.printInfo();
    }
}
